tests/ignite/handlers/test_time_limit.py

```diff
@@ -0,0 +1,31 @@
+import time
+
+import pytest
+
+from ignite.engine import Engine, Events
+from ignite.handlers import TimeLimit
+
+
+def test_arg_validation():
+
+    with pytest.raises(ValueError, match=r"Argument limit_sec should be a positive integer."):
+        TimeLimit(limit_sec=-5)
+
+    with pytest.raises(TypeError, match=r"Argument limit_sec should be an integer."):
+        TimeLimit(limit_sec="abc")
+
+
+def test_terminate_on_time_limit():
+    def _train_func(engine, batch):
+        time.sleep(1)
+
+    def _test(n_iters, limit):
```

Comment:
Maybe we should check that `Events.TERMINATE` is triggered

```
trainer.add_event_handler(Events.ITERATION_COMPLETED, TimeLimit(limit))
trainer.state.is_terminated = False
@trainer.on(Events.TERMINATE)
def _():
    trainer.state.is_terminated = True
trainer.run(range(n_iters))
assert trainer.state.is_terminated == (n_iters > limit)
```
What do you think ?

Reply:
Yes, make sense, thanks !

Reply:
done